rvgo/test/syscall_test.go

```diff
@@ -30,22 +30,32 @@ func staticOracle(t *testing.T, preimageData []byte) *testOracle {
 	}
 }
 
-func runEVM(t *testing.T, contracts *Contracts, addrs *Addresses, stepWitness *fast.StepWitness, fastPost fast.StateWitness) {
+func runEVM(t *testing.T, contracts *Contracts, addrs *Addresses, stepWitness *fast.StepWitness, fastPost fast.StateWitness, revertCode []byte) {
 	env := newEVMEnv(t, contracts, addrs)
-	evmPost, _, _ := stepEVM(t, env, stepWitness, addrs, 0)
+	evmPost, _, _ := stepEVM(t, env, stepWitness, addrs, 0, revertCode)
 	require.Equal(t, hexutil.Bytes(fastPost).String(), hexutil.Bytes(evmPost).String(),
 		"fast VM produced different state than EVM")
 }
 
-func runSlow(t *testing.T, stepWitness *fast.StepWitness, fastPost fast.StateWitness, po slow.PreimageOracle) {
+func runSlow(t *testing.T, stepWitness *fast.StepWitness, fastPost fast.StateWitness, po slow.PreimageOracle, expectedErr interface{}) {
 	slowPostHash, err := slow.Step(stepWitness.EncodeStepInput(fast.LocalContext{}), po)
-	require.NoError(t, err)
-	fastPostHash, err := fastPost.StateHash()
-	require.NoError(t, err)
-	require.Equal(t, fastPostHash, slowPostHash, "fast VM produced different state than slow VM")
+	if expectedErr != nil {
+		require.ErrorAs(t, err, expectedErr)
+	} else {
+		fastPostHash, err := fastPost.StateHash()
```

Comment:
```suggestion
                require.NoError(t, err)
		fastPostHash, err := fastPost.StateHash()
```

We should check error from `slow.Step`.